### PR TITLE
✨ Source MongoDB Internal POC: Generate Test Data

### PR DESCRIPTION
**Pull request by jdpgrailsdev**
## What
* Generate test data

## How
* Add a CLI-based tool to insert documents into MongoDB

## Recommended reading order
1. `MongoDbInsertClient.kt`
2. `build.gradle`

### CODE REVIEW

**Comment by octavia-squidington-iii** [comment]
##  source-mongodb-internal-poc test report (commit [`64b3770ab6`](https://github.com/airbytehq/airbyte/pull/29049/commits/64b3770ab6c988a609c4ff570e05e93384b1419e)) - ❌

⏲️  Total pipeline duration: 5.09s 

| Step   | Result   |
|--------|----------|

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/jonathan_generate-mongo-test-data/1691076111/64b3770ab6c988a609c4ff570e05e93384b1419e/source-mongodb-internal-poc/0.0.1/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:64b3770ab6c988a609c4ff570e05e93384b1419e)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-mongodb-internal-poc test
```

**Comment by octavia-squidington-iii** [comment]
##  source-mongodb-internal-poc test report (commit [`327affe457`](https://github.com/airbytehq/airbyte/pull/29049/commits/327affe457351446b8e28104636ae729a1a7925c)) - ❌

⏲️  Total pipeline duration: 5.69s 

| Step   | Result   |
|--------|----------|

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/jonathan_generate-mongo-test-data/1691079874/327affe457351446b8e28104636ae729a1a7925c/source-mongodb-internal-poc/0.0.1/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:327affe457351446b8e28104636ae729a1a7925c)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-mongodb-internal-poc test
```

**Comment by octavia-squidington-iii** [comment]
##  source-mongodb-internal-poc test report (commit [`cf274ced23`](https://github.com/airbytehq/airbyte/pull/29049/commits/cf274ced23b68ed8a1de1fada265339aeb6541ce)) - ❌

⏲️  Total pipeline duration: 18.86s 

| Step   | Result   |
|--------|----------|

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/jonathan_generate-mongo-test-data/1691080503/cf274ced23b68ed8a1de1fada265339aeb6541ce/source-mongodb-internal-poc/0.0.1/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:cf274ced23b68ed8a1de1fada265339aeb6541ce)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-mongodb-internal-poc test
```

**Comment by octavia-squidington-iii** [comment]
##  source-mongodb-internal-poc test report (commit [`750c1788dc`](https://github.com/airbytehq/airbyte/pull/29049/commits/750c1788dc497a46d2b3ac696701e3cabec8c183)) - ❌

⏲️  Total pipeline duration: 18.94s 

| Step   | Result   |
|--------|----------|

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/jonathan_generate-mongo-test-data/1691080649/750c1788dc497a46d2b3ac696701e3cabec8c183/source-mongodb-internal-poc/0.0.1/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:750c1788dc497a46d2b3ac696701e3cabec8c183)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-mongodb-internal-poc test
```

**Comment by octavia-squidington-iii** [comment]
##  source-mongodb-internal-poc test report (commit [`d30665a8cd`](https://github.com/airbytehq/airbyte/pull/29049/commits/d30665a8cdacfdcbee27bdea38dc8d6b7cdb5bcf)) - ❌

⏲️  Total pipeline duration: 4.81s 

| Step   | Result   |
|--------|----------|

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/jonathan_generate-mongo-test-data/1691085514/d30665a8cdacfdcbee27bdea38dc8d6b7cdb5bcf/source-mongodb-internal-poc/0.0.1/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:d30665a8cdacfdcbee27bdea38dc8d6b7cdb5bcf)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-mongodb-internal-poc test
```

File: airbyte-integrations/connectors/source-mongodb-internal-poc/src/test/kotlin/MongoDbInsertClient.kt [hunk 1]
```diff
@@ -0,0 +1,52 @@
+package io.airbyte.integrations.source.mongodb.internal
+
+import io.airbyte.commons.json.Jsons
+import kotlinx.cli.ArgParser
```
Comment (colesnodgrass) [inline airbyte-integrations/connectors/source-mongodb-internal-poc/src/test/kotlin/MongoDbInsertClient.kt:4]: TIL

File: airbyte-integrations/connectors/source-mongodb-internal-poc/src/test/kotlin/MongoDbInsertClient.kt [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        val databaseName by parser.option(ArgType.String, fullName = "database-name", shortName = "d", description = "Database Name").required()
+        val collectionName by parser.option(ArgType.String, fullName = "collection-name", shortName = "cn", description = "Collection Name").required()
+        val replicaSet by parser.option(ArgType.String, fullName = "replica-set", shortName = "r", description = "Replica Set").required()
+        val username by parser.option(ArgType.String, fullName = "username", shortName = "u", description = "Username").required()
+        val numberOfDocuments by parser.option(ArgType.Int, fullName = "number", shortName = "n", description = "Number of documents to generate").default(10000)
+
+        parser.parse(args)
+
+        println("Enter password: ")
+        var password = readln()
```
Comment (colesnodgrass) [inline airbyte-integrations/connectors/source-mongodb-internal-poc/src/test/kotlin/MongoDbInsertClient.kt:27]: Should this be `val`?

File: airbyte-integrations/connectors/source-mongodb-internal-poc/src/test/kotlin/MongoDbInsertClient.kt [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+        var config = mapOf(MongoConstants.DATABASE_CONFIGURATION_KEY to databaseName,
```
Comment (colesnodgrass) [inline airbyte-integrations/connectors/source-mongodb-internal-poc/src/test/kotlin/MongoDbInsertClient.kt:29]: Should this be `val`?

**Review by colesnodgrass — APPROVED** [review]

**Comment by octavia-squidington-iii** [comment]
##  source-mongodb-internal-poc test report (commit [`6583bd09d2`](https://github.com/airbytehq/airbyte/pull/29049/commits/6583bd09d2aabc58e251840a93d72c8629036b83)) - ❌

⏲️  Total pipeline duration: 5.22s 

| Step   | Result   |
|--------|----------|

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/jonathan_generate-mongo-test-data/1691087076/6583bd09d2aabc58e251840a93d72c8629036b83/source-mongodb-internal-poc/0.0.1/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:6583bd09d2aabc58e251840a93d72c8629036b83)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-mongodb-internal-poc test
```

**Comment by octavia-squidington-iii** [comment]
##  source-mongodb-internal-poc test report (commit [`4a2ddda302`](https://github.com/airbytehq/airbyte/pull/29049/commits/4a2ddda30299c72c97713b28f4f0f8133d897107)) - ❌

⏲️  Total pipeline duration: 18.64s 

| Step   | Result   |
|--------|----------|

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/jonathan_generate-mongo-test-data/1691087700/4a2ddda30299c72c97713b28f4f0f8133d897107/source-mongodb-internal-poc/0.0.1/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:4a2ddda30299c72c97713b28f4f0f8133d897107)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-mongodb-internal-poc test
```

**Comment by octavia-approvington** [comment]
Approve this <br/> imagine the commander saying yes

**Review by octavia-approvington — APPROVED** [review]

**Pull request merged**